src/lib.rs

```diff
@@ -228,14 +233,16 @@ fn read_methods<'a>(
         if !is_method_descriptor(&descriptor) {
             fail!("Invalid descriptor for class method {}", i);
         }
-        if allow_init && name == "<init>" && !descriptor.ends_with('V') {
+        let descriptor = MethodDescriptor::parse(&descriptor)?;
```

Comment:
Can we drop the `is_method_descriptor` call above and roll that into the parse? The parse should fail if it's not valid so this is effectively doing the validation twice now.

Ditto for the field descriptor.